src/moleculesampleapp/VectorPoint.java

```diff
@@ -0,0 +1,43 @@
+package moleculesampleapp;
+
+import com.sun.javafx.geom.Vec3d;
+import javafx.scene.paint.Color;
+
+/**
+ * Created by Piotr on 06/02/2016.
+ */
+public class VectorPoint {
+    public Vec3d position;
```

Comment:
should be final. 

http://programmers.stackexchange.com/a/50230 - efective java item 14 (the polish version of this book currently have @ggajoch)

Java didn't insert safepoints on field access and there could be problems on GC.